From 6b28a3119f558e9096ba8f02578fcdd2f86bba4b Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 4 Jun 2022 18:59:13 +0200
Subject: [PATCH] NFC disabled by default and fix no catch exception

Two fixes for the NFC application:
- NFC disabled by default
- Fix Security exception
---
 src/com/android/nfc/NfcRootActivity.java | 5 +++--
 src/com/android/nfc/NfcService.java      | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/com/android/nfc/NfcRootActivity.java b/src/com/android/nfc/NfcRootActivity.java
index cc216f2..2a3131b 100644
--- a/src/com/android/nfc/NfcRootActivity.java
+++ b/src/com/android/nfc/NfcRootActivity.java
@@ -36,8 +36,9 @@ public class NfcRootActivity extends Activity {
             if (launchIntent != null) {
                 try {
                     startActivityAsUser(launchIntent, UserHandle.CURRENT);
-                } catch (ActivityNotFoundException e) {
-                }
+                 } catch (ActivityNotFoundException e) {
+                 } catch (SecurityException e) {
+                 }
             }
         }
         finish();
diff --git a/src/com/android/nfc/NfcService.java b/src/com/android/nfc/NfcService.java
index 837940c..75c0179 100644
--- a/src/com/android/nfc/NfcService.java
+++ b/src/com/android/nfc/NfcService.java
@@ -123,7 +123,7 @@ public class NfcService implements DeviceHostListener {
     public static final String PREF = "NfcServicePrefs";
 
     static final String PREF_NFC_ON = "nfc_on";
-    static final boolean NFC_ON_DEFAULT = true;
+    static final boolean NFC_ON_DEFAULT = false;
     static final String PREF_NDEF_PUSH_ON = "ndef_push_on";
     static final boolean NDEF_PUSH_ON_DEFAULT = false;
     static final String PREF_SECURE_NFC_ON = "secure_nfc_on";
-- 
2.30.2

